activesupport/lib/active_support/core_ext/hash/compact.rb

```diff
@@ -1,23 +1,27 @@
 class Hash
-  # Returns a hash with non +nil+ values.
-  #
-  #   hash = { a: true, b: false, c: nil }
-  #   hash.compact        # => { a: true, b: false }
-  #   hash                # => { a: true, b: false, c: nil }
-  #   { c: nil }.compact  # => {}
-  #   { c: true }.compact # => { c: true }
-  def compact
-    select { |_, value| !value.nil? }
+  unless self.instance_methods(false).include?(:compact)
```

Comment:
This should be an explicit check on `Hash.instance_methods(false)`, not sure how the behaviour would be with `Hash` inherited classes.